### PR TITLE
[decorators] Add support for initializers

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
tc39/proposal-decorators@cc2db18b5215c91fc86b02a03f20506b7c9c6365

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      | yes
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
TODO: I forgot to update the helper version

File: packages/babel-helpers/src/helpers.js [hunk 1]
```diff
@@ -1358,17 +1363,27 @@ helpers.decorate = helper("7.1.5")`
...
+      if (
+        element.kind === "method" &&
+        (placement === "static" || placement === "prototype")
+      ) {
+        var receiver = placement === "static" ? F : proto;
+        _defineClassElement(receiver, element);
+      }
+    });
+
+    elements.forEach(function(element /*: ElementDescriptor */) {
```
Comment (danez) [inline packages/babel-helpers/src/helpers.js]: Why two loops instead of combining the two?

Comment (nicolo-ribaudo) [reply packages/babel-helpers/src/helpers.js]: Becose they differ both in the if condition and in the behavior inside the if.

Comment (nicolo-ribaudo) [reply packages/babel-helpers/src/helpers.js]: I just noticed that I misinterpreted your question 🙃

The proposal says that all methods must be defined before the fields/initializers. If I used a single loop, that order wouldn't be maintained.

**Comment by rushjoe18** [comment]
Thanks

**Comment by littledan** [comment]
The new test of ordering looks good to me. I'd say, this patch is ready to land.

**Comment by nicolo-ribaudo** [comment]
When is the next meeting? I can enable it using a flag in the meantime

**Comment by nicolo-ribaudo** [comment]
We discussed this at the meeting today and decided to wait for the TC39 meeting, since we don't want to introduce a flag which we might deprecate in two weeks.

**Comment by nicolo-ribaudo** [comment]
@littledan 	What is the meeting outcome?

**Comment by nicolo-ribaudo** [comment]
Blocked by tc39/proposal-decorators#182

**Comment by littledan** [comment]
@nicolo-ribaudo We seem to be in agreement about adding the feature, but a name bikeshed remains.

**Comment by nicolo-ribaudo** [comment]
The feature is now enabled behind an option: `["@babel/plugin-proposal-decorators", { "initializers": true }]`, thanks to #9166. That way, if the name will change it will be easy to update Babel :smile:

**Comment by nicolo-ribaudo** [comment]
I will open a different PR for "hooks"

**Pull request closed without merging**